README.md

```diff
@@ -1,173 +1,175 @@
+
+To exit an Erdstall network with all assets, call `ErdstallSession.leave()`. This will first send an `exit` request to the Erdstall enclave. Once the epoch has ended, an exit proof will be received from the operator. Due to protocol-security-related reasons, it now has to be waited for another full epoch before an on-chain `withdraw` transaction can be sent to the Erdstall contract, containing the exit proof that was received in the prior epoch. The call returns a list of transaction promises. Once all those transactions have been mined, the user's assets will be withdrawn from Erdstall back into their Ethereum wallet.
+
+### Erdstall transactions
+
+The following transactions are currently possible in any Erdstall network.
+
+#### Transfers
+
+To transfer assets within Erdstall, use `Erdstall.transferTo(assets, to)`.
+This transfers the specified assets to the specified recipient, and returns a transaction receipt.
+
+#### Trading
+
+Erdstall natively supports (basket) trading of tokens and NFTs. Currently, simple two-step offer and accept trading is supported. Some external marketplace infrastructure needs to be in place to present trade offers to potential buyers, who can then execute a trade on the Erdstall network.
+
+A trade offer can be created using `ErdstallSession.createOffer()`. The returned `TradeOffer` needs the external marketplace infrastructure to reach potential buyers. An interested buyer can then execute the trade by calling `ErdstallSession.acceptTrade(offer)`, which will send a `Trade` transaction to the Erdstall network, atomically exchanging the assets between the two accounts.
+
+#### NFT minting & burning
+
+`ErdstallSession.mint(token, id)` can be used to mint a new NFT with on-chain contract address `token` and id `id`. The NFT can then be traded freely and is only minted on-chain upon withdrawal.
+
+NFT-owners can burn their tokens with a `ErdstallSession.burn(nfts)` transaction call. If the NFT was deposited into Erdstall, it will not be burned on-chain, but will indefinitely be locked-up and be held by the NFT-Holder contract.
+
+## Event Subscriptions & Callbacks
+
+After creating an Erdstall client, event handlers can be set up using `Erdstall.on()` and `Erdstall.once()`, like
+
+```ts
+erdClient.on("receipt", (rec) => {
+	console.log(`Transaction by ${rec.tx.sender}`);
+});
+```
+
+Once all event handlers are setup, first `await Erdstall.initialize()` to initialize the client, then `await Erdstall.subscribe()` to subscribe to all balance proofs and transaction receipts for all users, or filter to a specific user with address `who` by calling `Erdstall.subscribe(who)`. `ErdstallSession.subscribeSelf()` is a shortcut for subscribing a session to the own user.
```

Comment:
IMO this suggests that all event handlers have to be registered **before** initializing the Erdstall client which is not true, since one can of course register handlers afterwards as well.

Reply:
Added more explanation that this is also possible.